### PR TITLE
BFD-3698: Address missing global EC2 permissions

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
BFD-3698

### What Does This PR Do?
Historically, we've just ignored AMI provisioning errors and EC2 instance initialization errors. This seeks to address the consistent errors that result from insufficient IAM privileges within the primary instance profiles through the following:
1. Consolidate packer-specific IAM resources in `mgmt` under newly introduced `packer-iam.tf`
2. Define and import previously non-terraform managed packer resources into the `mgmt` state through this _packer-iam.tf_ file
  **NOTE**: ~These have since been removed from the state and must be re-imported,~ These have been imported into the `mgmt` state, i.e. `terraform import aws_iam_role.packer bfd-packer` and `terraform import aws_iam_instance_profile.packer bfd-packer`
4. Remove outdated, irrelevant policies from the packer role
5. Establish the `bfd-mgmt-ec2-instance-tags-ro` policy to globally allow _ec2:DescribeInstances_ and _ec2:DescribeTags_
6. Attach and/or embed the _instance-tags-ro_ policy to the relevant packer, migrator, pipeline, and server instance profile role definitions
7. Add an instruction to reconfigure Grub in the platinum definition. This is out-of-scope of BFD-3689, however, the validation couldn't proceed without it. Yielding to the reviewers to determine if this shouldn't belong elsewhere.

### What Should Reviewers Watch For?
* Do these changes seem reasonable?
* Do they make sense?
* Do any FIXME comments still exist within the change set?

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Have you fully verified and tested these changes? Is the acceptance criteria met? Please provide reproducible testing instructions, code snippets, or screenshots as applicable.
 
* I successfully provisioned platinum and application images with these changes using Jenkins.
* Out-of-band, I applied those images to the `test` environment. 
* I witnessed the expected drop in what have become _expected_ failures during AMI provisioning and EC2 initialization:
  * The recent v2.166.0 release yielded 32 distinct, assumed roles by ec2 instances with one or more failures
  * Provisioning and initialization with this branch into the `test` environment yielded zero.

### CODE REVIEW

File: ops/terraform/env/mgmt/packer-iam.tf [hunk 1]
```diff
@@ -0,0 +1,120 @@
+resource "aws_iam_policy" "packer_s3" {
+  #FIXME: This policy should be reconsidered before acceptance of BFD-3698. Suggest removal.
```
Comment (mjburling) [inline ops/terraform/env/mgmt/packer-iam.tf:2]: `bfd-mgmt-packer-s3` was moved here as part of the consolidation effort, however, it probably shouldn't exist...

File: ops/terraform/env/mgmt/packer-iam.tf [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      Version = "2012-10-17"
+    }
+  )
+  description           = "Allows EC2 instances to call AWS services on your behalf."
+  force_detach_policies = false
+  managed_policy_arns = [
+    aws_iam_policy.packer_ssm.arn,
+    aws_iam_policy.packer_s3.arn,
+    aws_iam_policy.packer_kms.arn,
+    "arn:aws:iam::aws:policy/AmazonElasticFileSystemFullAccess", #FIXME: This should be reconsidered before acceptance of BFD-3698. Suggest removal.
```
Comment (mjburling) [inline ops/terraform/env/mgmt/packer-iam.tf:108]: This AWS Managed policy is _probably_ a hold-over from either earlier EFT efforts and/or self-hosted Jenkins systems that relied on EFS for multi-mount capabilities between Jenkins systems and ephemeral, Packer-managed EC2 instances. This should probably be removed.

File: ops/terraform/env/mgmt/README.md [hunk 3]
```diff
@@ -157,7 +160,7 @@ The management or `mgmt` environment is home to some higher-order resources that
 | [aws_ssm_parameter.bcda_aws_account_number](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameter) | data source |
 | [aws_ssm_parameter.cbc_aws_account_arn](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameter) | data source |
 | [aws_ssm_parameter.cpm_aws_account_arn](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameter) | data source |
-| [aws_ssm_parameters_by_path.common_sensitive](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameters_by_path) | data source |
+| [aws_ssm_parameters_by_path.params](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameters_by_path) | data source |
```
Comment (mjburling) [inline ops/terraform/env/mgmt/README.md:163]: This is unrelated to BFD-3698. Probably outdated readme from recent updates to the configuration management strategy where we neglected to regenerated terraform documentation.

File: ops/terraform/env/mgmt/iam.tf [hunk 4]
```diff
@@ -1,90 +1,25 @@
-#TODO: Determine if the bfd-packages sees continued use
-resource "aws_iam_policy" "packer_s3" {
```
Comment (mjburling) [inline ops/terraform/env/mgmt/iam.tf:2]: This policy definition was fully removed. Answering the above TODO: the `bfd-packages` S3 bucket at issue does not exist and has not existed any time recently.

File: ops/packer/scripts/platinum/03-install-security-updates.sh [hunk 5]
```diff
@@ -2,3 +2,6 @@
 
 # Apply security patches
 sudo yum update-minimal --security -y
+
+# Aggressively reconfigure grub configuration
+sudo grub2-mkconfig -o /boot/grub2/grub.cfg
```
Comment (mjburling) [inline ops/packer/scripts/platinum/03-install-security-updates.sh:7]: We might consider leaving this out of #2493 because it is strictly orthogonal. However, as implemented, this works.

File: ops/terraform/env/mgmt/packer-iam.tf [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+{
+  "Statement": [
+    {
+      "Action": [
+        "ssm:GetParametersByPath",
+        "ssm:GetParameters",
+        "ssm:GetParameter"
+      ],
+      "Effect": "Allow",
+      "Resource": [
```
Comment (timothy-ellis-ky) [inline ops/terraform/env/mgmt/packer-iam.tf:15]: recommend use of ${local.region} in place of "us-east-1"

**Review by timothy-ellis-ky — COMMENTED** [review]
see comment for env/mgmt/packer-iam.tf

**Review by timothy-ellis-ky — APPROVED** [review]

File: ops/packer/scripts/platinum/03-install-security-updates.sh (hunk 5 above)
Comment (malessi) [reply ops/packer/scripts/platinum/03-install-security-updates.sh:7]: Honestly, I'm perfectly OK including it here. In my opinion, it's not worth creating a new ticket for 1 line, and we want to get our Platinums working again ASAP anyways.
